test/inductor/test_cache.py

```diff
@@ -0,0 +1,150 @@
+# Owner(s): ["module: inductor"]
+import random
+import time
+
+import torch
+import torch._dynamo
+import torch._inductor.config as inductor_config
+import torch.nn as nn
+import torch.nn.functional as F
+from torch._dynamo.test_case import TestCase
+
+from torch.testing._internal.common_utils import set_rng_seed, TEST_WITH_ASAN
+from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+
+aten = torch.ops.aten
+
+
+class Foo(nn.Module):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.linear = nn.Linear(15, 30)
+
+    def forward(self, x: torch.Tensor):
+        y = x * x
+        y = self.linear(y)
+        y += 3
+        y -= 1
+        return y
+
+
+used_cache_keys = []
+
+
+def activate_unique_cache():
+    new_cache_key = random.randint(0, 1000000000)
+    while new_cache_key in used_cache_keys:
+        new_cache_key = random.randint(0, 1000000000)
+    used_cache_keys.append(new_cache_key)
+    inductor_config.fx_graph_cache = new_cache_key
+
+
+class CacheTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls._saved_config = inductor_config.save_config()
+
+    def tearDown(self):
+        super().tearDown()
+        inductor_config.load_config(self._saved_config)
+
+    def setUp(self):
+        super().setUp()
+        # We don't want a fixed seed for this test to avoid cache collisions
+        set_rng_seed(int(time.time()))
```

Comment:
same here, really try to make the tests deterministic, and figure out some other way to deal with collisions.